File: src/features/typing/data/ts/subsets.ts
function subsets(nums: number[]): number[][] {
    const result: number[][] = [];
    const backtrack = (start: number, current: number[]): void => {
        result.push([...current]);
        for (let i = start; i < nums.length; i++) {
            backtrack(i + 1, [...current, nums[i]]);
        }
    };
    backtrack(0, []);
    return result;
}

File: src/features/typing/data/dart/subsets.dart
class Solution {
  List<List<int>> subsets(List<int> nums) {
    final result = <List<int>>[];
    void backtrack(int start, List<int> current) {
      result.add(List<int>.of(current));
      for (int i = start; i < nums.length; i++) {
        current.add(nums[i]);
        backtrack(i + 1, current);
        current.removeLast();
      }
    }
    backtrack(0, <int>[]);
    return result;
  }
}
